database/upload.go

```diff
@@ -15,6 +15,7 @@ import (
 type Upload struct {
 	ID        primitive.ObjectID `bson:"_id,omitempty" json:"id"`
 	UserID    primitive.ObjectID `bson:"user_id,omitempty" json:"userId"`
+	IP        string             `bson:"ip" json:"IP"`
```

Comment:
I think `uploader_ip` is a more descriptive name for this field.

Reply:
I find that a bit redundant but fine, no harm in that.